src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4124,14 +4130,12 @@ class StubGenerator: public StubCodeGenerator {
     Register state0    =  x5; // t0
     Register state1    =  x6; // t1
     Register state2    =  x7; // t2
-    Register state3    =  x9; // s1
+    Register state3    = x20; // s4
```

Comment:
I don't think it's safe to use x5 (t0) / x6 (t1) to keep some long-lived values like `state0` / `state1`. Those two are reserved scratch registers which could be explictly / implicitly clobberred by various assembler functions like [1]. 

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/macroAssembler_riscv.hpp#L760-L763

Reply:
Yeah, I tried to make sure that I pass any temp registers to such functions explicitly, but you're right, it can be error prone. Should I just use s4 -> s7 for the 4 state registers?

Reply:
Hmmm... Maybe I don't need to declare / use `tmp3` and just use `t0` in `add_u32`? Sounds reasonable?